STAR-Delegation/draft-ietf-acme-star-delegation.md

```diff
@@ -603,19 +619,74 @@ certificate has been issued by the CA, the IdO:
 At this point of the protocol flow, the same considerations as in
 {{sec-cname-installation}} apply.
 
+Before forwarding the Order request to the CA, the IdO SHOULD ensure that the
+selected CA supports "unauthenticated GET" by inspecting the relevant settings
+in the CA's `directory` object, as per {{sec-nego-allow-cert-get}}.  If the CA
+does not support "unauthenticated GET" of certificate resources, the IdO MUST
+NOT forward the Order request.  Instead, it MUST move the Order status to
+`invalid` and set the `allow-certificate-get` to `false`.  The same occurs in
+case the Order request is forwarded and the CA does not reflect the
+`allow-certificate-get` setting in its Order resource.  The combination of
+`invalid` status and denied `allow-certificate-get` in the Order resource at
+the IdO provides an unambiguous (asynchronous) signal to the NDC about the
+failure reason.
+
 ### Capability Discovery
```

Comment:
Sorry, we're not done yet (or I'm getting more and more confused):

* Is the `meta` object on the IdO always formatted "the new way", without `auto-renewal`, even if it is used for delegation of STAR certs?
* Where are we saying that the CA (not the IdO) must use `meta` "the new way" if it is used to delegate non-STAR certs?

More generally to the second bullet: most of the document discusses the NDC-IdO leg. But capability negotiation also happens on the IdO-CA leg, so we need to describe this leg somewhere. See my new issue #187.

Reply:
> * Is the `meta` object on the IdO always formatted "the new way", without `auto-renewal`, even if it is used for delegation of STAR certs?

yes, IdO's directory meta only has "delegation-enabled".  "auto-renewal" is only for the CA.

> * Where are we saying that the CA (not the IdO) must use `meta` "the new way" if it is used to delegate non-STAR certs?

in the "Capability Discovery" section:
```
In order to help a client to discover support for certificate fetching using
unauthenticated HTTP GET, the directory object of an ACME server (typically,
one deployed by the CA) contains the following attribute in the `meta` field:

- allow-certificate-get (optional, boolean): See {{sec-nego-allow-cert-get}}.
```

> More generally to the second bullet: most of the document discusses the NDC-IdO leg. But capability negotiation also happens on the IdO-CA leg, so we need to describe this leg somewhere. See my new issue #187.

I am very happy with any clarification, but isn't the "Capability Discovery" text sufficient?
```
In order to help a client to discover support for this profile, the directory
object of an ACME server (typically, one deployed by the IdO) contains the
following attribute in the `meta` field:

- delegation-enabled (optional, boolean): Boolean flag indicating support for
  the profile specified in this memo.  An ACME server that supports this
  delegation profile MUST include this key, and MUST set it to true.

The `delegation-enabled` flag may be specified regardless of the existence or
setting of the `auto-renewal` flag.
```

(It looks like the last para needs some tweaking.)

Reply:
Suggest to rewrite the last paragraph:
> The IdO MUST declare its support for delegation using `delegation-enabled` regardless of whether it supports delegation of STAR certificates, non-STAR certificates or both.

Reply:
done in 03cbe7b